### ISSUE TITLE
Define possible actions when fixers cannot be safely applied

### ISSUE THREAD

**Issue by ivanbasov** [dotnet/roslyn-analyzers#1458]
For example, here we should trigger a warning:
```
class C
{
    void M(int x) { } // CA1801: ReviewUnusedParameters
    
    int N(int x) => x;
    
    void Caller()
    {
        M(N(0));
    }
}
```

However, trying to apply it, we found that the caller executes invokes a method  as a parameter to M. Therefore, we cannot remove the caller method. How should we respond to UI? The current idea is to just not apply the fix could also confuse users.

**Comment by ivanbasov** [comment]
Discussed this on a design meeting. Decided to make the fix even if it can bring side effects.
FYI: @mavasani, @jinujoseph, @kuhlenh
